### PR TITLE
Problem: prost and tonic upgrade fails (fixes #303)

### LINKED ISSUE #303: Problem: prost and tonic upgrade fails
> May need to wait for https://github.com/cosmos/cosmos-rust/pull/195. Since private trait MessageExt only implements for [prost::Message of prost-0.9.0](https://github.com/cosmos/cosmos-rust/blob/main/cosmrs/src/prost_ext.rs#L14).

### PR DESCRIPTION
**Pull request by tomtau**
Solution:
- bump all proto-related crates (cosmrs, ibc-rs, prost...)
- regenerate protos with the new tonic build
- replace grpc-web-client with tonic-web-wasm-client
- adjust IBC message (new timeout height is an enum)

### CODE REVIEW

File: proto/src/prost/chainmain.nft.v1.rs [hunk 1]
```diff
@@ -1,101 +1,102 @@
...
     #![allow(unused_variables, dead_code, missing_docs, clippy::let_unit_value)]
     use tonic::codegen::*;
-    #[doc = " Msg defines the NFT Msg service."]
+    /// Msg defines the NFT Msg service.
     #[derive(Debug, Clone)]
     pub struct MsgClient<T> {
         inner: tonic::client::Grpc<T>,
     }
-    // Workaround, add feature manually, it could be fixed after https://github.com/hyperium/tonic/issues/491
-    #[cfg(feature = "transport")]
```
Comment (damoncro) [inline proto/src/prost/chainmain.nft.v1.rs:96]: This is needed to added back. It is a workaround.

File: proto/src/prost/chainmain.nft.v1.rs [hunk 2]
```diff
@@ -108,292 +109,319 @@ pub mod msg_client {
...
     #![allow(unused_variables, dead_code, missing_docs, clippy::let_unit_value)]
     use tonic::codegen::*;
-    #[doc = " Query defines the gRPC querier service for NFT module"]
+    /// Query defines the gRPC querier service for NFT module
     #[derive(Debug, Clone)]
     pub struct QueryClient<T> {
         inner: tonic::client::Grpc<T>,
     }
-    // Workaround, add feature manually, it could be fixed after https://github.com/hyperium/tonic/issues/491
-    #[cfg(feature = "transport")]
```
Comment (damoncro) [inline proto/src/prost/chainmain.nft.v1.rs:394]: Same

File: proto/src/prost/chainmain.supply.v1.rs [hunk 3]
```diff
@@ -2,36 +2,33 @@
...
     #![allow(unused_variables, dead_code, missing_docs, clippy::let_unit_value)]
     use tonic::codegen::*;
-    #[doc = " Query defines the gRPC querier service."]
+    /// Query defines the gRPC querier service.
     #[derive(Debug, Clone)]
     pub struct QueryClient<T> {
         inner: tonic::client::Grpc<T>,
     }
-    // Workaround, add feature manually, it could be fixed after https://github.com/hyperium/tonic/issues/491
-    #[cfg(feature = "transport")]
```
Comment (damoncro) [inline proto/src/prost/chainmain.supply.v1.rs:32]: Same

File: proto-build/src/main.rs [hunk 4]
```diff
@@ -236,7 +235,6 @@ fn compile_luna_classic_protos_and_services(out_dir: &Path) {
     tonic_build::configure()
         .build_client(true)
         .build_server(false)
-        .format(true)
```
Comment (damoncro) [inline proto-build/src/main.rs:239]: any reasons to remove format?

File: proto/src/prost/chainmain.nft.v1.rs [hunk 5]
```diff
@@ -108,292 +111,321 @@ pub mod msg_client {
...
+    #[prost(string, tag="4")]
     pub data: ::prost::alloc::string::String,
-    #[prost(string, tag = "5")]
+    #[prost(string, tag="5")]
     pub owner: ::prost::alloc::string::String,
 }
 /// Denom defines a type of NFT
-#[derive(Serialize, Deserialize, Clone, PartialEq, ::prost::Message)]
+#[derive(Serialize, Deserialize)]
+#[derive(Clone, PartialEq, ::prost::Message)]
```
Comment (damoncro) [inline proto/src/prost/chainmain.nft.v1.rs:275]: Is it because no formating?

File: proto-build/src/main.rs (hunk 4 above)
Comment (tomtau) [reply proto-build/src/main.rs:239]: It doesn't exist in 0.7 tonic-build: https://docs.rs/tonic-build/0.7.2/tonic_build/struct.Builder.html#implementations
See: https://github.com/hyperium/tonic/commit/d6c0fc112b2288a080fd0a727453b24d666e3d79

File: proto/src/prost/chainmain.nft.v1.rs (hunk 5 above)
Comment (tomtau) [reply proto/src/prost/chainmain.nft.v1.rs:275]: that's the output I got after running `make proto`

Comment (tomtau) [reply proto/src/prost/chainmain.nft.v1.rs:275]: there's formatting using https://github.com/dtolnay/prettyplease

Comment (tomtau) [reply proto/src/prost/chainmain.nft.v1.rs:275]: > This library is designed using the simplest possible algorithm and data structures that can deliver about 95% of the quality of rustfmt-formatted output.

**Review by damoncro — APPROVED** [review]

File: common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs [hunk 6]
```diff
@@ -301,7 +303,7 @@ pub enum CosmosRawNormalMsg {
         /// the tokens to be transferred
         token: SingleCoin,
         /// Timeout height relative to the current block height.
-        /// The timeout is disabled when set to 0.
+        /// The timeout is disabled when set to Never.
```
Comment (leejw51crypto) [inline common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs:306]: where `Never` declared?

**Review by leejw51crypto — APPROVED** [review]
lgtm

File: common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs (hunk 6 above)
Comment (tomtau) [reply common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs:306]: oh, that's a variant in `TimeoutHeight`, I switched back to height, but didn't updated the comment

Comment (tomtau) [reply common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs]: good catch!

**Pull request merged**